src/algokit_utils/applications/app_factory.py

```diff
@@ -0,0 +1,791 @@
+            deletable=params.deletable if params.deletable is not None else self._factory._deletable,
+            deploy_time_params=(
+                params.deploy_time_params
+                if params.deploy_time_params is not None
+                else self._factory._deploy_time_params
+            ),
+        )
+
+        compiled = self._factory.compile(
+            AppClientCompilationParams(
+                deploy_time_params=create_params.deploy_time_params,
+                updatable=create_params.updatable,
+                deletable=create_params.deletable,
+            )
+        )
+
+        result = self._factory._handle_call_errors(
+            lambda: self._factory._parse_method_call_return(
+                lambda: self._algorand.send.app_create_method_call(self._factory.params.create(create_params)),
+                self._factory._app_spec.get_arc56_method(params.method).to_abi_method(),
+            )
+        )
+
+        return (
+            self._factory.get_app_client_by_id(
+                app_id=result.app_id,
+            ),
+            AppFactoryCreateMethodCallResult(
+                transaction=result.transaction,
+                confirmation=result.confirmation,
+                tx_id=result.tx_id,
+                app_id=result.app_id,
+                app_address=result.app_address,
+                abi_return=result.abi_return,
+                compiled_approval=compiled.compiled_approval if compiled else None,
+                compiled_clear=compiled.compiled_clear if compiled else None,
+                group_id=result.group_id,
+                tx_ids=result.tx_ids,
+                transactions=result.transactions,
+                confirmations=result.confirmations,
+                returns=result.returns,
+            ),
+        )
+
+
+class AppFactory:
+    def __init__(self, params: AppFactoryParams) -> None:
+        self._app_spec = AppClient.normalise_app_spec(params.app_spec)
+        self._app_name = params.app_name or self._app_spec.name
+        self._algorand = params.algorand
+        self._version = params.version or "1.0"
+        self._default_sender = params.default_sender
+        self._default_signer = params.default_signer
+        self._deploy_time_params = params.deploy_time_params
+        self._updatable = params.updatable
+        self._deletable = params.deletable
+        self._approval_source_map: SourceMap | None = None
+        self._clear_source_map: SourceMap | None = None
+        self._params_accessor = _AppFactoryParamsAccessor(self)
+        self._send_accessor = _AppFactorySendAccessor(self)
+        self._create_transaction_accessor = _AppFactoryCreateTransactionAccessor(self)
+
+    @property
+    def app_name(self) -> str:
+        return self._app_name
+
+    @property
+    def app_spec(self) -> Arc56Contract:
+        return self._app_spec
+
+    @property
+    def algorand(self) -> AlgorandClientProtocol:
```

Comment:
```suggestion
    def algorand(self) -> AlgorandClient:
```
We don't want to use an interface/protocol at all, as it has annoying breaking change side effects when new fields are added. See https://github.com/algorandfoundation/algokit-utils-ts/pull/365

Reply:
✅